config_cmd.go

```diff
@@ -23,6 +23,12 @@ width: 80
 
 func defaultConfigFile() string {
 	scope := gap.NewScope(gap.User, "glow")
+	if c := os.Getenv("CHARM_CONFIG_HOME"); c != "" {
+		return filepath.Join(c, "glow.yml")
+	}
+	if c := os.Getenv("XDG_CONFIG_HOME"); c != "" {
```

Comment:
Why not using os.UseConfigDir ?

https://pkg.go.dev/os#UserConfigDir

It's implementation might be what you are looking for

https://cs.opensource.google/go/go/+/refs/tags/go1.22.5:src/os/file.go;l=509

Reply:
UserConfigDir also looks at the home dir `os.GetEnv("HOME")`  if `os.GetEnv("XDG_CONFIG_HOME")` is empty which isn't wanted here.

Reply:
Indeed

Reply:
> On Darwin, it returns $HOME/Library/Application Support

I think it wouldn't change much in that particular problem.

We can work on replacing the usage of gap with this library though

Reply:
Please note, I know some macOS users expect to use this folder by default, but want code to respect XDG_CONFIG_HOME if present.

- https://github.com/charmbracelet/glow/pull/594#issuecomment-2189627733

Reply:
yes, this is what this PR does...